analyzer/analyzer.go

```diff
@@ -291,3 +286,32 @@ func isDisabled(t Type, disabled []Type) bool {
 	}
 	return false
 }
+
+func MergeOsVersion(old, new *types.OS) *types.OS {
+	esmVersionSuffix := "ESM"
+	switch {
+	case old == nil:
+		return new
+	// OLE also has /etc/redhat-release and it detects OLE as RHEL by mistake.
+	// In that case, OS must be overwritten with the content of /etc/oracle-release.
+	// There is the same problem between Debian and Ubuntu.
+	case old.Family == aos.RedHat, old.Family == aos.Debian:
+		return new
+	// Ubuntu has ESM program: https://ubuntu.com/security/esm
+	// OS version and esm status are stored in different files.
+	// We have to merge OS version after parsing these files.
+	case old.Family == aos.Ubuntu && new.Family == aos.Ubuntu:
+		if strings.HasSuffix(old.Name, "-"+esmVersionSuffix) { // version number with ESM suffix already stored
+			return old
+		}
+		if old.Name == esmVersionSuffix {
+			new.Name = new.Name + "-" + old.Name
+			return new
+		}
+		if new.Name == esmVersionSuffix {
+			new.Name = old.Name + "-" + new.Name
+			return new
+		}
```

Comment:
It is a bit tricky. What if we add `Extended` field to `OS`? Then, we can use the field for Extended LTS (Debian), Extended Life Cycle Support (Red Hat), and so on.

Reply:
Done

Reply:
Done